cspp_runner/cspp_utils.py

```diff
@@ -0,0 +1,262 @@
+        # save time! Currently we check all files...
+        # FIXME!
+        update_it = True
+        for filename in files:
+            tslot = datetime.strptime(
+                os.path.basename(filename).split('.')[-1], '%Y%m%d%H%M')
+            if now - tslot < tdelta:
+                update_it = False
+                break
+
+        if not update_it:
+            LOG.info('An URL update trial has been attempted recently. Continue')
+            return True
+
+        LOG.info('No update trial seems to have been attempted recently')
+        tdelta = timedelta(days=int(self.thr_lut_files_age_days))
+
+        files_ok = True
+        LOG.info("Directory " + str(self.lut_dir) + "...")
+        files = glob(os.path.join(self.lut_dir, '*'))
+        if len(files) == 0:
+            LOG.info("No LUT files available!")
+            return False
+
+        filename = files[0]
+        tstamp = os.stat(filename)[stat.ST_MTIME]
+        first_time = datetime.utcfromtimestamp(tstamp)
+
+        if (now - first_time) > tdelta:
+            LOG.info("File too old! File=%s " % filename)
+            files_ok = False
+
+        return files_ok
+
+    def _update_lut_files(self, timeout=600):
+        """Update LUT files for VIIRS processing.
+
+        Updates the ancillary LUT files over internet.
+
+        These files need to be updated at least once every week, in order to
+        achieve the best possible SDR processing.
+
+        """
+        update_files(
+            self.url_jpss_remote_lut_dir,
+            self.lut_update_stampfile_prefix,
+            self.mirror_jpss_luts,
+            "LUT",
+            timeout=timeout)
+
+    def _update_ancillary_files(self, timeout=600):
+        """Update the dynamic ancillary files needed for CSPP VIIRS SDR processing.
+
+        These data files encompass Two Line Element (TLE) and Polar Wander (PW)
+        files, and should preferably be updated daily. This is done
+        automatically in CSPP if the viirs_sdr script is run without the '-l'
+        option. However, to slightly speed up the processing and avoid hangups
+        depending on internet connectivity this script can omit the automatic
+        download (using the '-l' option) and thus the files need to be updated
+        outside the script.
+
+        """
+        update_files(
+            self.url_jpss_remote_anc_dir,
+            self.anc_update_stampfile_prefix,
+            self.mirror_jpss_ancillary,
+            "ANC",
+            timeout=timeout)
+
+
+def update_files(url_jpss_remote_dir, update_stampfile_prefix, mirror_jpss,
+                 what, timeout=600):
+    """Do the update of the LUT files on disk.
+
+    Runs the fetch over internet from a separate working directory, and calls
+    the JPSS script in a separat shell.
+
+    """
+    _check_environment("CSPP_WORKDIR")
+    cspp_workdir = os.environ.get("CSPP_WORKDIR", '')
```

Comment:
```suggestion
    cspp_workdir = os.environ["CSPP_WORKDIR"]
```
see comment elsewhere